### PR TITLE
Move FuzzerSetRand to separate file.

### PR DESCRIPTION
**Pull request by richsalz**
This fixes Google's OSS-Fuzz builds.

### CODE REVIEW

**Comment by kcc** [comment]
I don't see how this is going to fix anything. 
You still use rand for "standalone" build but (looks like) don't build it when linking the actual fuzzer. 
Note that this logic is also somewhere in the openssl build system.
Here is how oss-fuzz builds the fuzzers: 
https://github.com/google/oss-fuzz/blob/master/projects/openssl/build.sh
(this build script assumes certain env variables, see https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md#buildsh-script-environment

**Comment by richsalz** [comment]
It's late, and I guess my mind is fuzzy, but are you assuming each fuzzer is just one file?  More from me tomorrow...

**Comment by richsalz** [comment]
oh yes, I guess so are.  So I have to ```#include``` rand.c ?

**Comment by kcc** [comment]
I assume that the openssl build system assumes that each fuzzer is just one file. 
Yes, `#include "rand.c"` will help, although I would rather make it `rand.inc` (up to you, of course)

**Comment by richsalz** [comment]
Yeah, rand.inc is what I did.  Additional commit pushed.

File: fuzz/rand.inc [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (paulidale) [inline fuzz/rand.inc]: Date?

**Comment by kroeckx** [comment]
You can just add dependencies in build.info. You could also move it to driver.c which is used by all binaries.

**Comment by richsalz** [comment]
Apparently not; their build script just compiles a single .c file and uses their own driver.

**Comment by richsalz** [comment]
merged, thanks!

**Comment by kroeckx** [comment]
No it doesn't, it just runs make but links in their library that provides main, which is exactly the same as how you set it up using libfuzzer. It needs to functions in driver.c, because that is what their main calls.

**Comment by richsalz** [comment]
so you are saying that rand.inc could be merged inline into driver.c?  @kcc is that true?

**Comment by kroeckx** [comment]
@richsalz: I wrote all that code, so I assume that I know what I'm talking about.

**Comment by richsalz** [comment]
I apologize, I meant no offense.  It's just that I understood slightly differently from the google folks.  It is probably my misunderstanding.  And it is definitely my fault for challenging you.

Do you want to see a new PR that merges the inline stuff into driver.c?

**Comment by kroeckx** [comment]
I assume it's going to work now, there hasn't been a new build, so I don't really care. But I don't like an ".inc" file, just name it ".h" if you want to include things. Or name it ".c" and change build.info saying in which files it should get linked in. So many options.

**Comment by kcc** [comment]
My local build succeeded, thanks! 
the automatic build should recover later today.

**Pull request closed without merging**